### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by cathook**
As title.

### CODE REVIEW

**Comment by cathook** [comment]
@dwi2

File: js/classic/ui.js [hunk 1]
```diff
@@ -10,42 +10,158 @@
   var ui = exports.classic.ui;
 
   /**
-   * An UI component represents a switch.
+   * Creates a DOM element and wraps it by a jquery selector.
    *
-   * It supports turing on/off operations.
+   * @param {String} nodeName - Name of the DOM element to be created.
+   * @param {String=} className - Additional classes to add if needs.
```
Comment (dwi2) [inline js/classic/ui.js]: Could you remove the '=' in `{String=}` ? jsdoc will failed to parse this tags

File: js/classic/ui.js [hunk 2]
```diff
@@ -10,42 +10,158 @@
...
+     */
+    destroy: function() {
+      this.selector.remove();
+    },
+
+    /**
+     * Calls an event handler.
+     *
+     * @param {String} handlerName - Name of the event handler.
+     * @param {=} args - Arguments to pass to the event handler.
```
Comment (dwi2) [inline js/classic/ui.js]: Could you remove the '=' in `{=}` ? jsdoc will failed to parse this tags.
You could either use `{}` or remove `{=}` completely.
That is to say:

```
@param args - Arguments to pass to the event handler.
```

or 

```
@param {} args - Arguments to pass to the event handler.
```

File: js/classic/ui.js (hunk 1 above)
Comment (cathook) [inline js/classic/ui.js]: Ok, but how can I represent that the parameter's type can be a `String` and `undefined`.

File: js/classic/ui.js (hunk 2 above)
Comment (cathook) [inline js/classic/ui.js]: Okey, I will fix it.

File: js/classic/ui.js (hunk 1 above)
Comment (dwi2) [inline js/classic/ui.js]: just use `String`, and  add description in the back says that it could sometimes be `undefined`

**Comment by cathook** [comment]
@dwi2

**Comment by dwi2** [comment]
LGTM :+1: 
[LGTM](http://lgtm.in/i/YK2se2bWp)

**Pull request merged**